src/libsyntax/ast.rs

```diff
@@ -456,7 +472,7 @@ pub enum MetaItemKind {
     /// List meta item.
     ///
     /// E.g. `derive(..)` as in `#[derive(..)]`
-    List(InternedString, Vec<P<MetaItem>>),
+    List(InternedString, Vec<MetaListItem>),
```

Comment:
Thinking about this: MetaItem almost always sites behind a P<> anyways, so there should be very little, if any, performance degradation. We might even see a performance win once literals start being use due to coalesced allocations.